src/components/AppHeader/AppHeader.tsx

```diff
@@ -52,8 +53,15 @@ const useStyles = makeStyles({
 
 export const AppHeader = () => {
   const classes = useStyles();
-  // const navigate = useNavigate();
   const userContext = useContext(UserContext);
+  const roles = useRoles();
+
+  const myRoles = roles.data?.filter(
+    (item) => Number(item.user.Id) === userContext.user.Id
+  );
```

Comment:
I think this might be better suited as a hook -- so that you can also use it for things like -- are they an OSF -- CSF -- etc

Reply:
Don't need to make it a network call -- just a hook that calls the hook you already have that is caching data, and returns an object in a format you can use, with the info you need.  Such as useMyRoles() -- it can internally call the useRoles hook and just apply the filter you wrote here but so that it can be used by any other component without having to rewrite the filter logic each time.  Thinking about it a little more, probably better than an array of the roles they are in, is an object of isAdmin, isOSF, isCSF, etc.  That takes all the possible roles and is a boolean of if they have the role or not.  That way you don't have to do the includes() call in each component either, just directly be able to write something like "if(myRoles.isAdmin)" or "myRoles.isAdmin &&"